src/cortex_cli/auth.py

```diff
@@ -78,13 +83,39 @@ def login_request(url: str, realm: str, client_id: str, username: str, password:
     result = requests.post(request_url, data=data.dict(exclude_none=True), timeout=AUTH_REQUESTS_TIMEOUT)
     if result.status_code == 404:
         raise ClientAuthenticationError(f'token endpoint is not available at {url}')
+    if result.status_code == 400 and result.json().get('error_description', '') == 'Account is not fully set up':
+        raise ClientAccountSetupError('Account is not fully set up')
     if result.status_code != 200:
         raise ClientAuthenticationError('invalid username and/or password')
     tokens = result.json()
     tokens = {key: tokens.get(key, '') for key in ['access_token', 'refresh_token']}
     return tokens
 
 
+def update_password(url: str, realm: str, username: str, password: str, new_password: str):
+    """Update temporary password using authentication server's account view.
+
+    This works only when account has a temporary password,
+    not to be used for changing permanent passwords.
+    """
+    browser = mechanize.Browser()
+    browser.set_handle_robots(False)
+    browser.open(f'{url}/realms/{realm}/account')
+
+    # Enter old credentials into login form
+    browser.select_form(nr=0)
+    browser['username'] = username
+    browser['password'] = password
+    browser.submit()
+
+    # Enter new credentials into password update form
+    browser.select_form(nr=0)
+    browser['password-new'] = new_password
+    browser['password-confirm'] = new_password
+    browser.submit()
```

Comment:
is there a way to check whether the submitted request succeeded before closing the browser?

Reply:
If submitting succeeds the account view should open. I added a check for that.